Robus/src/reception.c

```diff
@@ -77,34 +79,36 @@ void Recep_GetHeader(volatile unsigned char *data)
             printf("cmd : 0x%04x\n", current_msg->header.cmd);                 /*!< msg definition. */
             printf("size : 0x%04x\n", current_msg->header.size);               /*!< Size of the data field. */
 #endif
-            // Reset the catcher.
-            data_count = 0;
-            // Switch state machiine to data reception
-            ctx.data_cb = Recep_GetData;
-            // Cap size for big messages
-            if (current_msg->header.size > MAX_DATA_MSG_SIZE)
-            {
-                data_size = MAX_DATA_MSG_SIZE;
-            }
-            else
-            {
-                data_size = current_msg->header.size;
-            }
-            if (keep)
-            {
-                // start crc computation
-                LuosHAL_ComputeCRC((unsigned char *)current_msg->stream, sizeof(header_t), (unsigned char *)&crc_val);
-                if (data_size)
+                // Reset the catcher.
+                data_count = 0;
+                // Switch state machiine to data reception
+                ctx.data_cb = Recep_GetData;
+                // Cap size for big messages
+                if (current_msg->header.size > MAX_DATA_MSG_SIZE)
                 {
-                    MsgAlloc_ValidHeader();
+                    data_size = MAX_DATA_MSG_SIZE;
                 }
-            }
-            else
-            {
-                MsgAlloc_InvalidMsg();
-            }
-        }
+                else
+                {
+                    data_size = current_msg->header.size;
+                }
+                if (keep)
+                {
+                    if (data_size)
+                    {
+                        MsgAlloc_ValidHeader();
+                    }
+                }
+                else
+                {
+                    MsgAlloc_InvalidMsg();
+                }
+            break;
+
+        default:
+        break;
     }
+    LuosHAL_ComputeCRC((uint8_t *)data, 1, (uint8_t *)&crc_val);
```

Comment:
Because we have to check and reload context at each TX we have to compute only byte by byte. We should remove the size.